src/mjolnir/ferry_connections.cc

```diff
@@ -177,7 +181,10 @@ uint32_t ShortestPath(const uint32_t start_node_idx,
         }
 
         // Add to the node labels and adjacency set. Skip if this is a loop.
-        node_labels.emplace_back(cost, endnode, expand_node_idx);
+        const bool remove_destonly =
+            first_edge_destonly ? (pred_destonly && w.destination_only()) : false;
```

Comment:
similarly here: only remove destonly if the pred had it too. it's still a bit too broad, but can't it any other way I think.